src/main/java/pocketpal/backend/storage/Storage.java

```diff
@@ -0,0 +1,227 @@
+// @@author nghochi123
+package pocketpal.backend.storage;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.math.RoundingMode;
+import java.text.DecimalFormat;
+import java.text.ParseException;
+import java.time.LocalDateTime;
+
+import pocketpal.backend.constants.Config;
+import pocketpal.backend.constants.MiscellaneousConstants;
+import pocketpal.data.entry.Category;
+import pocketpal.data.entry.Entry;
+import pocketpal.frontend.util.CategoryUtil;
+import pocketpal.frontend.util.DateTimeUtil;
+import pocketpal.frontend.exceptions.InvalidCategoryException;
+import pocketpal.frontend.exceptions.InvalidDateException;
+import pocketpal.backend.exceptions.InvalidReadFileException;
+
+public class Storage {
+    private final String filePath;
+    private final String delimiter;
+    private final DecimalFormat decimalFormat = new DecimalFormat("0.00");
+
+    public Storage() {
+        this(Config.RELATIVE_FILE_NAME);
+    }
+
+    /**
+     * Alternative constructor for Storage that allows the changing of filePath,
+     * used for Testing purposes only
+     *
+     * @param filePath Path to file that serialized entries will be stored in
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+        this.delimiter = Config.DELIMITER;
+    }
+
+    /**
+     * Creates a file and its respective parent directories if the file does
+     * not exist, using a predefined file path.
+     *
+     * @throws IOException If an error occurs in the creation of the new file
+     */
+    private void makeFileIfNotExists() throws IOException {
+        File file = new File(this.filePath);
+        file.getParentFile().mkdirs();
+        file.createNewFile();
+        new FileOutputStream(
```

Comment:
Do not need to start new line for each input parameter. If the line is too long use line wrapping at appropriate places of the line. Indentation for wrapped lines should be 8 spaces